packages/web-components/gulp-tasks/vendor.js

```diff
@@ -1,165 +0,0 @@
-/**
- * @license
- *
- * Copyright IBM Corp. 2020, 2023
- *
- * This source code is licensed under the Apache-2.0 license found in the
- * LICENSE file in the root directory of this source tree.
- */
-
-'use strict';
-
-const gulp = require('gulp');
-const {
-  carbonWebComponentsCJSSrcDir,
-  carbonWebComponentsESSrcDir,
-  carbonWebComponentsVendorSrcDir,
-  carbonWebComponentsVendorCJSDstDir,
-  carbonWebComponentsVendorESDstDir,
-  servicesCJSSrcDir,
-  servicesESSrcDir,
-  servicesVendorSrcDir,
-  servicesVendorCJSDstDir,
-  servicesVendorESDstDir,
-  servicesStoreCJSSrcDir,
-  servicesStoreESSrcDir,
-  servicesStoreVendorSrcDir,
-  servicesStoreVendorCJSDstDir,
-  servicesStoreVendorESDstDir,
-  utilitiesCJSSrcDir,
-  utilitiesESSrcDir,
-  utilitiesVendorSrcDir,
-  utilitiesVendorCJSDstDir,
-  utilitiesVendorESDstDir,
-} = require('./config');
-
-/**
- * Generates `src/internal/vendor` contents.
- */
-const carbonWebComponentsVendorSrc = () =>
-  gulp
-    .src([`${carbonWebComponentsESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(carbonWebComponentsVendorSrcDir));
-
-/**
- * Generate `es/internal/vendor` contents.
- */
-const carbonWebComponentsVendorESDst = () =>
-  gulp
-    .src([`${carbonWebComponentsESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(carbonWebComponentsVendorESDstDir));
-
-/**
- * Generate `lib/internal/vendor` contents.
- */
-const carbonWebComponentsVendorCJSDst = () =>
-  gulp
-    .src([`${carbonWebComponentsCJSSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(carbonWebComponentsVendorCJSDstDir));
-
-/**
- * Generates `src/internal/vendor` contents.
- */
-const servicesVendorSrc = () =>
-  gulp
-    .src([`${servicesESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesVendorSrcDir));
-
-/**
- * Generate `es/internal/vendor` contents.
- */
-const servicesVendorESDst = () =>
-  gulp
-    .src([`${servicesESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesVendorESDstDir));
-
-/**
- * Generate `lib/internal/vendor` contents.
- */
-const servicesVendorCJSDst = () =>
-  gulp
-    .src([`${servicesCJSSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesVendorCJSDstDir));
-
-/**
- * Generates `src/internal/vendor` contents.
- */
-const servicesStoreVendorSrc = () =>
-  gulp
-    .src([`${servicesStoreESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesStoreVendorSrcDir));
-
-/**
- * Generate `es/internal/vendor` contents.
- */
-const servicesStoreVendorESDst = () =>
-  gulp
-    .src([`${servicesStoreESSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesStoreVendorESDstDir));
-
-/**
- * Generate `lib/internal/vendor` contents.
- */
-const servicesStoreVendorCJSDst = () =>
-  gulp
-    .src([`${servicesStoreCJSSrcDir}/**/*`, '!**/*-{test,story}.js'])
-    .pipe(gulp.dest(servicesStoreVendorCJSDstDir));
```

Comment:
May have gone too far with removing this one? This copies the files from `packages/services-store/lib` into `packages/web-components/lib/internal/vendor/@carbon/ibmdotcom-services-store`. Similar with the one above, but `s/lib/es`. This appears to be a way of exposing the modules from `@carbon/ibmdotcom-services-store` to downstream users of `@carbon/ibmdotcom-web-components`, since the former is a private package that isn't published. 

This change would be a breaking change for any downstream users who might be using those imports.

Reply:
Yep, will probably want this back. CAEM is case and point on this. Although we only use it once, and only for type info, it does get used: https://github.ibm.com/webstandards/carbon-for-aem/blob/main/src/components/video-player/video-player-container.ts#L22

Reply:
I've added back the `vendor` build step, and left anything that copies source files into the `es/` or `lib/` directories, since that forms part of the public API of this package.